<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro" name="example">

    <ros2_control name="GazeboSystem" type="system">
        <hardware>
            <plugin>gazebo_ros2_control/GazeboSystem</plugin>
        </hardware>
        <joint name="joint1">
            <command_interface name="position">
                <param name="min">0.0</param>
                <param name="max">2.3</param>
            </command_interface>
            <command_interface name="velocity"/>
            <state_interface name="position">
                <param name="initial_value">0.0</param>
            </state_interface>
            <state_interface name="velocity"/>
            <state_interface name="effort"/>
        </joint>
        <joint name="joint2">
            <command_interface name="velocity"/>
            <command_interface name="position">
                <param name="min">0.0</param>
                <param name="max">1.2</param>
            </command_interface>
            <state_interface name="position">
                <param name="initial_value">0.0</param>
            </state_interface>
            <state_interface name="velocity"/>
            <state_interface name="effort"/>
        </joint>
        <joint name="joint3">
            <command_interface name="velocity"/>
            <command_interface name="position">
                <param name="min">-1.57</param>
                <param name="max">0</param>
            </command_interface>
            <state_interface name="position">
                <param name="initial_value">0.0</param>
            </state_interface>
            <state_interface name="velocity"/>
            <state_interface name="effort"/>
        </joint>
        <joint name="joint4">
            <command_interface name="velocity"/>
            <command_interface name="position">
                <param name="min">0.0</param>
                <param name="max">0.3</param>
            </command_interface>
            <state_interface name="position">
                <param name="initial_value">0.0</param>
            </state_interface>
            <state_interface name="velocity"/>
            <state_interface name="effort"/>
        </joint>
        <joint name="left_gear_joint">
            <command_interface name="position">
                <param name="min">-1</param>
                <param name="max">-0.4</param>
            </command_interface>
            <command_interface name="velocity"/>
            <state_interface name="position">
                <param name="initial_value">0.0</param>
            </state_interface>
            <state_interface name="velocity"/>
            <command_interface name="effort"/>
            <state_interface name="effort"/>
        </joint>
        <joint name="left_finger_joint">
            <command_interface name="position">
                <param name="min">-0.5</param>
                <param name="max">0.5</param>
            </command_interface>
            <command_interface name="velocity"/>
            <state_interface name="position">
                <param name="initial_value">0.0</param>
            </state_interface>
            <state_interface name="velocity"/>
            <command_interface name="effort"/>
            <state_interface name="effort"/>
        </joint>
        <joint name="right_gear_joint">
            <command_interface name="position">
                <param name="min">0.4</param>
                <param name="max">1</param>
            </command_interface>
            <command_interface name="velocity"/>
            <state_interface name="position">
                <param name="initial_value">0.0</param>
            </state_interface>
            <state_interface name="velocity"/>
            <command_interface name="effort"/>
            <state_interface name="effort"/>
        </joint>
        <joint name="right_finger_joint">
            <command_interface name="position">
                <param name="min">-0.5</param>
                <param name="max">0.5</param>
            </command_interface>
            <command_interface name="velocity"/>
            <state_interface name="position">
                <param name="initial_value">0.0</param>
            </state_interface>
            <state_interface name="velocity"/>
            <command_interface name="effort"/>
            <state_interface name="effort"/>
        </joint>
    </ros2_control>

    <gazebo>
        <plugin filename="libgazebo_ros2_control.so" name="gazebo_ros2_control">
            <robot_sim_type>gazebo_ros2_control/GazeboSystem</robot_sim_type>
            <parameters>$(find robot_arm1)/config/controller.yaml</parameters>
        </plugin>
    </gazebo>

    <gazebo>
        <plugin name="grasp_plugin" filename="libgazebo_ros_grasp_plugin.so">
            <robotNamespace>/</robotNamespace>
            <bodyName>left_finger_link</bodyName>
            <attachName>right_finger_link</attachName>
        </plugin>
    </gazebo>

    <gazebo reference="left_finger_link">
        <mu1>1.5</mu1>
        <mu2>1.5</mu2>
        <material>Gazebo/Black</material>
    </gazebo>

    <gazebo reference="right_finger_link">
        <mu1>1.5</mu1>
        <mu2>1.5</mu2>
        <material>Gazebo/Black</material>
    </gazebo>

    <gazebo reference="base_link">
        <mu1>0.01</mu1>
        <mu2>0.01</mu2>
    </gazebo>

</robot>